supervisor/conf/config-2.yml

```diff
@@ -0,0 +1,89 @@
+# Connection details for the OpAMP server.
+opampServer:
+    endpoint: https://localhost:1729
+    tls:
+        # Client-side TLS configuration to connect to the OpAMP server.
+
+agents:
+    # Many collector related configuration is omitted in this section as its
+    # tailor-made supervisor can fill in with the defaults.
```

Comment:
It is not clear what the intent is. Is the Supervisor supposed to supervise more than one agent simultaneously? Why is it necessary?

Reply:
So, here's where I wanted to add some more clarity. I am assuming that we want to the supervisor to be generic enough that it is able to manage things other than the collector (like Fluentbit). This can be accomplished in two ways:

One option, as you suggest, is to implement the supervisor as composable pieces which can be wired together. This way we can build a new supervisor executable for a collector or fluentbit. 

Another option is to implement the supervisor as composable pieces just like the above option but allow several supervisor implementations to exist within the same executable and running as separate Go routines, for example. The individual supervisor implementation will still be the same as option one except that more than one will reside in the same artifact. The advantage, in my understanding, is that if you have a system where, say, both collector and fluentbit are running, you need only one supervisor executable that manages it. Internally, that executable will spawn two Go routines one managing collector and the other managing the fluentbit by communicating with one or more OpAMP servers.

Reply:
I think this complicates the Supervisor for a use case that is likely going to be rare (multiple agents on one machine). Even when you have multiple agents on one machine I don't think it is a big deal for each agent to have its own Supervisor. Let's keep it simple for now. The Supervisor if implemented correctly should be very lightweight, so we can afford to run one Supervisor instance per Agent instance.